docs/tileentities/tileentity.md

```diff
@@ -0,0 +1,112 @@
+
+!!! note
+	The method to register a `TileEntity` is using a `String` instead of a `ResourceLocation` before Forge version `14.23.3.2694`
+	You should still use the `ResourceLocation` format `modid:tileentity` though.
+	Not using this format will result in your entitiy being called `minecraft:tileentity`
+
+## Attaching a `TileEntity` to a `Block`
+To attach your new `TileEntity` to a `Block` you need to override 2 (two) methods within the Block class.
+```JAVA
+	Block#hasTileEntity(IBlockstate state)
+
+	Block#createTileEntity(World world, IBlockState state)
+```
+Using the parameters you can choose if the block should have a `TileEntity` or not.
+But usually you will return `true` in the first method and a new instance of your `TileEntity` in the second method.
+
+At this point the `TileEntity`can be used for Rendering.
+
+## Storing Data within your `TileEntity`
+In order to save data, create the following two methods
+```JAVA
+	TileEntity#writeToNBT(NBTTagCompound nbt):NBTTagCompound
+
+	TileEntity#readFromNBT(NBTTagCompound nbt)
+```
+This methods are called whenever the `TileEntity` gets (un-)loaded, you don't use them to access your data.
+All your data should be fields within you class, and be (de-)serialized through this methods.
+
+Whenever your data changes you need to call ```TileEntity#markDirty()```, otherwise your `TileEntity` might be skipped while saving.
+
+!!! note
+	It is important that you call the super methods!
+	Due to this following tag names are already occupied `id`,`x`,`y`,`z`,`ForgeData`,`ForgeCaps`.
+
+## Keeping a `TileEntity` through changing `BlockStates`
+There might be situations in which you need to change your `BlockState`, an example for this is the vanilla furnace,
+which changes its state from `lit=false` to `lit=true` when fuel and something smetlable is inside.
+Achieving this is rather simple, by overriding following method
+```JAVA
+	TileEntity#shouldRefresh(World world, BlockPos pos, IBlockState oldState, IBlockState newSate)#boolean
+```
+!!! note
+	You should actually check the `BlockStates` and not just return `false` in order to prevent unwanted behavior and bugs.
+    
+## Ticking `TileEntities`
+If you need a ticking `TileEntity` for example to keep track of the progress during a smelting process.
+You need to add the `ITickable` interface to your `TileEntity`.
+Now you can implement all your calculations within
+```JAVA
+	ITickable#update()
+```
+!!! note
+	This method is called each tick (usually 20 (twenty) times a second) you should avoid having complicated calculations in here.
+	And if possible, you should make more complex calculations just every X ticks.
+	(The amount of ticks in a second may be lower then 20 (twenty) but won't be higher)
+    
+## Syncing the Data to the Client
+There are 2 (two) ways of syncing data to the client.
+* Syncing on World Load
+* Syncing on changes
```

Comment:
This list is rather pointless. I think this should be split into three sections: Synchronizing on chunk load, Synchronizing on block updates and Synchronizing with a custom message.

For the third option: Maybe add a link to SimpleImpl here, and explicitly mention the `sendToAllTracking(IMessage, NetworkRegistry.TargetPoint)` method that is particularly useful for this.